tests/unit/api/c/receiver/test_receiver.cpp

```diff
@@ -66,6 +67,109 @@ class TestReceiver : public ::testing::Test
+          EXPECT_NE(ip_type, kEtcPalIpTypeInvalid);
+          EXPECT_EQ(universe, CHANGE_UNIVERSE_WORKS_SECOND_UNIVERSE);
+          EXPECT_NE(get_recv_thread_context(thread_id), nullptr);
+          EXPECT_NE(socket, nullptr);
+          return kEtcPalErrOk;
+        };
+        break;
+      case kSacnIpV6Only:
+        sacn_remove_receiver_socket_fake.custom_fake = [](sacn_thread_id_t thread_id, etcpal_socket_t* socket, bool) {
+          ASSERT_NE(socket, nullptr);
+          EXPECT_NE(*socket, CHANGE_UNIVERSE_WORKS_FIRST_IPV4_SOCKET);
+          EXPECT_NE(get_recv_thread_context(thread_id), nullptr);
+        };
+        sacn_add_receiver_socket_fake.custom_fake = [](sacn_thread_id_t thread_id, etcpal_iptype_t ip_type,
+                                                       uint16_t universe, const EtcPalMcastNetintId*, size_t,
+                                                       etcpal_socket_t* socket) {
+          EXPECT_NE(ip_type, kEtcPalIpTypeV4);
+          EXPECT_NE(ip_type, kEtcPalIpTypeInvalid);
+          EXPECT_EQ(universe, CHANGE_UNIVERSE_WORKS_SECOND_UNIVERSE);
+          EXPECT_NE(get_recv_thread_context(thread_id), nullptr);
+          EXPECT_NE(socket, nullptr);
+          return kEtcPalErrOk;
+        };
+        break;
+      case kSacnIpV4AndIpV6:
+        sacn_remove_receiver_socket_fake.custom_fake = [](sacn_thread_id_t thread_id, etcpal_socket_t* socket, bool) {
+          ASSERT_NE(socket, nullptr);
+          EXPECT_NE(get_recv_thread_context(thread_id), nullptr);
+        };
+        sacn_add_receiver_socket_fake.custom_fake = [](sacn_thread_id_t thread_id, etcpal_iptype_t ip_type,
+                                                       uint16_t universe, const EtcPalMcastNetintId*, size_t,
+                                                       etcpal_socket_t* socket) {
+          EXPECT_NE(ip_type, kEtcPalIpTypeInvalid);
+          EXPECT_EQ(universe, CHANGE_UNIVERSE_WORKS_SECOND_UNIVERSE);
+          EXPECT_NE(get_recv_thread_context(thread_id), nullptr);
+          EXPECT_NE(socket, nullptr);
+          return kEtcPalErrOk;
+        };
+    }
+
+    etcpal_error_t change_universe_result =
+        sacn_receiver_change_universe(handle, CHANGE_UNIVERSE_WORKS_SECOND_UNIVERSE);
+
+    EXPECT_EQ(change_universe_result, kEtcPalErrOk);
+    EXPECT_EQ(sacn_lock_fake.call_count, sacn_unlock_fake.call_count);
+    EXPECT_EQ(sacn_initialized_fake.call_count, 2u);
+    EXPECT_EQ(clear_term_set_list_fake.call_count, 1u);
+
+    switch (ip_supported)
+    {
+      case kSacnIpV4Only:
+      case kSacnIpV6Only:
+        EXPECT_EQ(sacn_remove_receiver_socket_fake.call_count, 1u);
+        EXPECT_EQ(sacn_add_receiver_socket_fake.call_count, 2u);
+        break;
+      case kSacnIpV4AndIpV6:
+        EXPECT_EQ(sacn_remove_receiver_socket_fake.call_count, 2u);
+        EXPECT_EQ(sacn_add_receiver_socket_fake.call_count, 4u);
+    }
+
+    clear_term_set_list_fake.custom_fake = nullptr;
```

Comment:
If these fakes are not going to be used again before the test is over, this isn't necessary - sacn_source_loss_reset_all_fakes() in the SetUp() method will take care of it.